### PR TITLE
ADCM-6211: Add tests for Cluster/Service/Component public API

### PR DESCRIPTION
**Pull request by DanBalalan**

### CODE REVIEW

File: adcm_aio_client/core/host_groups/action_group.py [hunk 1]
```diff
@@ -33,6 +34,12 @@ def actions(self: Self) -> ActionsAccessor:
 
 class ActionHostGroupNode(HostGroupNode[Union["Cluster", "Service", "Component"], ActionHostGroup]):
     class_type = ActionHostGroup
+    filtering = Filtering(
+        FilterBy("id", COMMON_OPERATIONS, int),
+        FilterByName,
+        FilterBy("owner_id", {"eq"}, int),
+        FilterBy("owner_type", {"eq"}, str),
```
Comment (Sealwing) [inline adcm_aio_client/core/host_groups/action_group.py:41]: should discuss it with @a-alferov

File: tests/integration/test_cluster.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+from adcm_aio_client.core.objects.cm import Bundle, Cluster
+from tests.integration.bundle import pack_bundle
+from tests.integration.conftest import BUNDLES
+
+pytestmark = [pytest.mark.asyncio]
+
+
+def _get_httpx_client_base_url(adcm_client: ADCMClient) -> tuple[AsyncClient, str]:
+    httpx_client = adcm_client._requester._client  # pyright: ignore[reportAttributeAccessIssue]
+    base_url = urljoin(str(adcm_client._requester._client.base_url), "api/v2/")  # pyright: ignore[reportAttributeAccessIssue]
```
Comment (Sealwing) [inline tests/integration/test_cluster.py:23]: can you please elaborate on this requester using approach: motivation, profits, etc.?

File: tests/integration/test_cluster.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    assert await _get_ansible_forks(adcm_client, cluster) == initial_ansible_forks + 5
+
+    new_name = "New cluster name"
+    await _update_cluster_name(adcm_client, cluster, new_name)
+    assert cluster.name != new_name
+    await cluster.refresh()
+    assert cluster.name == new_name
+
+
+async def test_cluster_api(
```
Comment (Sealwing) [inline tests/integration/test_cluster.py:220]: @a-alferov we need to discuss order and naming in tests

File: tests/integration/test_cluster.py (hunk 2 above)
Comment (DanBalalan) [reply tests/integration/test_cluster.py:23]: it is used for checking responses from ADCM / updating entities to test `refresh()` behavior. Just got existing client from fixtures. Are you suggesting creating new httpx_client for this?

File: adcm_aio_client/core/host_groups/action_group.py (hunk 1 above)
Comment (DanBalalan) [reply adcm_aio_client/core/host_groups/action_group.py]: removed `owner_id,` `owner_type` filters from host_groups

File: tests/integration/test_cluster.py (hunk 3 above)
Comment (DanBalalan) [reply tests/integration/test_cluster.py]: refactored

File: tests/integration/test_cluster.py (hunk 2 above)
Comment (DanBalalan) [reply tests/integration/test_cluster.py]: replaced with `conftest.httpx_client` fixture

**Comment by Starovoitov** [comment]
Shouldn't ADCM-6140 be the branch for merge?

File: adcm_aio_client/core/host_groups/action_group.py [hunk 4]
```diff
@@ -33,6 +34,7 @@ def actions(self: Self) -> ActionsAccessor:
 
 class ActionHostGroupNode(HostGroupNode[Union["Cluster", "Service", "Component"], ActionHostGroup]):
     class_type = ActionHostGroup
+    filtering = Filtering(FilterBy("id", COMMON_OPERATIONS, int), FilterByName)
```
Comment (Sealwing) [inline adcm_aio_client/core/host_groups/action_group.py:37]: why filtering by id?

File: adcm_aio_client/core/objects/cm.py [hunk 5]
```diff
@@ -115,8 +115,8 @@ def signature_status(self: Self) -> Literal["invalid", "valid", "absent"]:
     def _type(self: Self) -> Literal["cluster", "provider"]:
         return self._data["mainPrototype"]["type"]
 
-    @property
-    def license(self: Self) -> License:
+    @async_cached_property
+    async def license(self: Self) -> License:
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:119]: no await in here, something forgotten?

or just for similarity with Service API?

File: adcm_aio_client/core/objects/cm.py [hunk 6]
```diff
@@ -257,9 +259,10 @@ def cluster(self: Self) -> Cluster:
     def components(self: Self) -> "ComponentsNode":
         return ComponentsNode(parent=self, path=(*self.get_own_path(), "components"), requester=self._requester)
 
-    @property
-    def license(self: Self) -> License:
-        return License(self._requester, self._data)
+    @async_cached_property
+    async def license(self: Self) -> License:
+        prototype_data = (await self.requester.get(f"prototypes/{self._data["prototype"]["id"]}/")).as_dict()
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:264]: pls use `.get("prototypes", self._data[], ...)` without building string url

File: tests/integration/test_service.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    """
+    Cluster with 52 services, one not added
+    """
+
+    bundle_folder = tmp_path / "".join(random.sample(string.ascii_letters, k=6)).lower()
+    config_yaml_path = bundle_folder / "config.yaml"
+    create_yaml(data=prepare_bundle_data(), path=config_yaml_path)
+
+    with (bundle_folder / "service_license.txt").open("wt") as service_license_file:
+        service_license_file.write("By using this test bundle, you agreeing to write tests well\n")
```
Comment (Sealwing) [inline tests/integration/test_service.py:76]: you can use `write_text` directly on Path object

Comment (DanBalalan) [reply tests/integration/test_service.py]: fixed

File: adcm_aio_client/core/objects/cm.py (hunk 6 above)
Comment (DanBalalan) [reply adcm_aio_client/core/objects/cm.py]: fixed

File: adcm_aio_client/core/objects/cm.py (hunk 5 above)
Comment (DanBalalan) [reply adcm_aio_client/core/objects/cm.py:119]: yes, for .license() calls consistency

File: adcm_aio_client/core/host_groups/action_group.py (hunk 4 above)
Comment (DanBalalan) [reply adcm_aio_client/core/host_groups/action_group.py]: it was in wiki. removed

**Pull request closed without merging**